test-framework/test-suites/integration/tests/add/test_add_host.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+@pytest.mark.usefixtures("revert_database")
+class TestAddHost:
+
+	HOST_POSITIVE_TEST_DATA = [
+		('backend-0-0','','add_host_with_standard_naming_convention.json',),
+		('backend_4_3','appliance=backend rack=4 rank=3','add_host_with_appliance_rack_rank.json',),
+		('backend-7-7','appliance=backend rack=5 rank=5','add_host_with_standard_naming_and_appliance_rack_rank.json',)
+		]
+
+	@pytest.mark.parametrize("host_name, add_params, output_file", HOST_POSITIVE_TEST_DATA)
+	def test_add_host_positive_behaviour(self, host, host_name, add_params, output_file):
+		dirn = '/export/test-files/add/'
+		expected_output = open(dirn + output_file).read()
+
+		result = host.run(f'stack add host {host_name} {add_params}')
+		assert result.rc == 0
+
+		result = host.run(f'stack list host {host_name} output-format=json')
+		assert result.rc == 0
+
+		assert self.validate_add_host_positive_behaviour_json(result.stdout, expected_output)
+
+
```

Comment:
trim excess line breaks in this file down to 1 or 2 breaks between functions.  We're not super picky, by 5 is too many ;)

If you want to create logical breaks between groups of functions, put a comment or something instead.